test/spec/params.spec.ts

```diff
@@ -146,6 +146,158 @@ describe('params reached to function', () => {
+    attach(
+      'check headers from function',
+      `expected to include: ${['custom-header', customHeader]}\n actual: ${JSON.stringify(
+        data?.headers
+      )}`,
+      ContentType.TEXT
+    )
+    console.log(data?.headers)
+    expect(
+      (data?.headers as [Array<string>]).filter(([k, v]) => k === 'x-region' && v === validRegion)
+        .length > 0
+    ).toBe(true)
+  })
+
+  test('invoke with region overrides region in the client', async () => {
+    /**
+     * @feature headers
+     */
+    log('create FunctionsClient')
+    const fclient = new FunctionsClient(`http://localhost:${relay.container.getMappedPort(8081)}`, {
+      region: FunctionRegion.ApNortheast1,
+    })
+
+    log('invoke mirror')
+    const customHeader = nanoid()
+    const validRegion = FunctionRegion.ApSoutheast1
+
+    const { data, error } = await fclient.invoke<MirrorResponse>('mirror', {
+      headers: {
+        'custom-header': customHeader,
+        Authorization: `Bearer ${apiKey}`,
+      },
+      region: validRegion,
+    })
+
+    log('assert no error')
+    const expected = {
+      url: 'http://localhost:8000/mirror',
+      method: 'POST',
+      headers: data?.headers ?? [],
+      body: '',
+    }
+    expect(data).toEqual(expected)
+    attach(
+      'check headers from function',
+      `expected to include: ${['custom-header', customHeader]}\n actual: ${JSON.stringify(
+        data?.headers
+      )}`,
+      ContentType.TEXT
+    )
+    console.log(data?.headers)
+    expect(
+      (data?.headers as [Array<string>]).filter(([k, v]) => k === 'x-region' && v === validRegion)
+        .length > 0
+    ).toBe(true)
+  })
+
+  test('invoke with region overrides region in the client', async () => {
+    /**
+     * @feature headers
+     */
+    log('create FunctionsClient')
+    const validRegion = FunctionRegion.ApSoutheast1
+    const fclient = new FunctionsClient(`http://localhost:${relay.container.getMappedPort(8081)}`, {
+      region: validRegion,
+    })
+
+    log('invoke mirror')
+    const customHeader = nanoid()
+
+    const { data, error } = await fclient.invoke<MirrorResponse>('mirror', {
+      headers: {
+        'custom-header': customHeader,
+        Authorization: `Bearer ${apiKey}`,
+      },
+    })
+
+    log('assert no error')
+    const expected = {
+      url: 'http://localhost:8000/mirror',
+      method: 'POST',
+      headers: data?.headers ?? [],
+      body: '',
+    }
+    expect(data).toEqual(expected)
+    attach(
+      'check headers from function',
+      `expected to include: ${['custom-header', customHeader]}\n actual: ${JSON.stringify(
+        data?.headers
+      )}`,
+      ContentType.TEXT
+    )
+    console.log(data?.headers)
+    expect(
+      (data?.headers as [Array<string>]).filter(([k, v]) => k === 'x-region' && v === validRegion)
+        .length > 0
+    ).toBe(true)
+  })
+
+  test('invoke mirror with invoke header and valid region', async () => {
```

Comment:
@mansueli also this one duplicates the first one (line 149). 

wdyt about adding a new test that will check that if you set some valid region on the client level (constructor), but then override it on the invoke level to "any", there will be no header in the request?

Reply:
Done:
https://github.com/supabase/functions-js/pull/75/files#diff-8cfc42c763162f3b42260d46eef32a3bb8ba2e0cccadf486353af403526946e6R233